src/epics/chatEpics.js

```diff
@@ -0,0 +1,210 @@
+                const reader = new FileReader();
+                reader.addEventListener("loadend", () => {
+                    const authToken = reader.result;
+                    observer.next(authToken);
+                    observer.complete();
+                });
+
+                reader.readAsText(fileBlob);
+            })
+    })
+}
+
+function getAuthRequestSettings({ mID }) {
+    const settings = {
+        url: authUrl,
+        responseType: 'blob',
+        headers: {
+            'X-Minilock-Id': mID
+        },
+        method: 'GET'
+    }
+    return settings;
+}
+
+const initConnectionEpic = (action$) =>
+    action$.ofType(CHAT_INIT_CONNECTION)
+        .mergeMap(createKeyPairObservable)
+        .mergeMap(({ keyPair, mID, isNewRoom, passphrase }) => 
+            Observable.merge(
+                Observable.of(alertSuccess('New room created.'))
+                    .filter(() => isNewRoom)
+                    .do(() => document.location.hash = '#' + passphrase),
+            
+                ajax(getAuthRequestSettings({ mID }))
+                    .map(ajaxResult => ({
+                        message: ajaxResult.response,
+                        mID,
+                        secretKey: keyPair.secretKey
+                    }))
+                    .mergeMap(createDecryptMessageObservable)
+                    .mergeMap(authToken =>
+                        chatHandler.initConnection({
+                            authToken,
+                            secretKey: keyPair.secretKey,
+                            mID
+                        })
+                    )
+                    .mergeMap(() =>
+                        Observable.merge(
+                            chatHandler.getMessageSubject()
+                            .map(addMessage),
+
+                            chatHandler.getUserStatusSubject()
+                            .map(setUserStatus),
+
+                            Observable.of(connectionInitiated())
+                        ))
+                
+            )
+        )
+        .catch(error => {
+            console.error(error);
+            return Observable.from([
+                alertWarning('Something is wrong with the chat thingie. Trying to reconnect...'),
+                disconnected()
+            ])
+        });
+
+const setUsernameEpic = (action$) =>
+    action$.ofType(CHAT_SET_USERNAME)
+    .filter(action => action.username)
+    .mergeMap(action => 
+        Observable.of(action.username)
+            .do(username => {
+                const ttl = USER_STATUS_DELAY_MS / 1000;
+                chatHandler.sendUserStatus(username, 'viewing', ttl)
+            })
+            .retryWhen(error => error.delay(500))
+    )
+    .map(username => usernameSet(username))
+
+const ownUserStatusEpic = (action$, store) =>
+    action$.ofType(CHAT_INIT_CHAT)
+    .mergeMap(() =>
+        createDetectVisibilityObservable()
+        //.throttleTime(USER_STATUS_DELAY_MS)
```

Comment:
@Spetastian Should we throttle here?  I'm not sure what triggers `ownUserStatusEpic` to run, so I'm not sure.